base/reflection.jl

```diff
@@ -1203,3 +1211,11 @@ REPL tab completion on `x.` shows only the `private=false` properties.
 propertynames(x) = fieldnames(typeof(x))
 propertynames(m::Module) = names(m)
 propertynames(x, private) = propertynames(x) # ignore private flag by default
+
+"""
+    hasproperty(x, s::Symbol)
+
+Returns a boolean indicating whether the object `x` has the specified property as one of
+its own properties.
+"""
+hasproperty(x, s::Symbol) = isdefined(x, s)
```

Comment:
This doesn't work for user-defined properties, unless someone overloads `hasproperty`.

`s in propertynames(x)` seems like a better (more correct, albeit slower) fallback.